redfish-core/lib/processor.hpp

```diff
@@ -975,6 +995,280 @@ inline void
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        functional = *value;
+                    }
+                }
+            }
+            else if (interface == "xyz.openbmc_project.Inventory.Item")
+            {
+                for (const auto& [proName, proValue] : properties)
+                {
+                    if (proName == "Present")
+                    {
+                        const bool* value = std::get_if<bool>(&proValue);
+                        if (value == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        present = *value;
+                    }
+                    else if (proName == "PrettyName")
+                    {
+                        const std::string* prettyName =
+                            std::get_if<std::string>(&proValue);
+                        if (prettyName == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        asyncResp->res.jsonValue["Name"] = *prettyName;
+                    }
+                }
+            }
+        }
+
+        if (!present)
+        {
+            asyncResp->res.jsonValue["Status"]["State"] = "Absent";
+        }
+        else
+        {
+            if (!functional)
+            {
+                asyncResp->res.jsonValue["Status"]["Health"] = "Critical";
+            }
+        }
+    }
+}
+
+inline void
+    getCpuCoreDataByService(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& service,
+                            const std::string& objPath)
+{
+    BMCWEB_LOG_DEBUG("Get available system cpu core resources by service.");
+
+    asyncResp->res.jsonValue["Status"]["State"] = "Enabled";
+    asyncResp->res.jsonValue["Status"]["Health"] = "OK";
+
+    dbus::utility::getManagedObjects(
+        service,
+        sdbusplus::message::object_path("/xyz/openbmc_project/inventory"),
+        std::bind_front(afterGetCpuCoreDataByService, asyncResp, objPath));
+}
+
+inline void aferGetSubProcessorData(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, const std::string& coreId,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreeResponse& subtree)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error, ec: {}", ec.value());
+        // No processor objects found by mapper
+        if (ec.value() == boost::system::errc::io_error)
+        {
+            messages::resourceNotFound(
+                asyncResp->res, "#Processor.v1_18_0.Processor", processorId);
+            return;
+        }
+
```

Comment:
Should we add a `ERROR` here?

Or, move L1161 to here as ERROR?

Reply:
if 1161 is an error, please it down past the resourcenotfound so the resourcenotfound doesn't log